queries.sql

```diff
@@ -38,4 +38,88 @@ WHERE name NOT LIKE 'Gabumon';
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * 
 FROM animals
-WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; 
\ No newline at end of file
+WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; 
+
+-- Inside a transaction update the animals table by setting the 
+-- species column to unspecified. Verify that change was made. 
+-- Then roll back the change and verify that species columns
+--  went back to the state before transaction.
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
+
+-- Update the animals table by setting the species column to 
+-- digimon for all animals that have a name ending in mon.
+BEGIN;
+UPDATE animals
+SET species = 'digmon'
+WHERE name LIKE '%mon';
+
+-- Update the animals table by setting the species column to
+-- pokemon for all animals that don't have species already set.
+BEGIN;
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL
+COMMIT;
```

Comment:
- [ ] Kindly verify that the change was made before and after the commit is made as stated in the requirements. e.g
```sql
BEGIN;
UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

SELECT species from animals; -- verify that change was made
COMMIT;
SELECT species from animals; -- verify that change persists after commit
```